pkg/bitcoin/hd.go

```diff
@@ -64,3 +72,47 @@ func (s *Service) DeriveExtendedKey(
 	response.ChainCode = xKey.ChainCode()
 	return response, nil
 }
+
+// Useful service to get keypair (xpub + privKey) from a seed for testing.
+// Random seed is generated if no seed is provided.
+func (s *Service) GetKeypair(seed string, chainParams ChainParams) (Keypair, error) {
+	var (
+		seedBytes []byte
+		response  Keypair
+	)
+
+	if seed == "" {
+		// Generate a random seed at the recommended length.
+		generatedSeed, err := hdkeychain.GenerateSeed(hdkeychain.RecommendedSeedLen)
+		if err != nil {
+			return response, err
+		}
+
+		seedBytes = generatedSeed
+
+	} else {
+		seedBytes = []byte(seed)
+	}
+
+	// Generate a new master node using the seed.
+	masterKey, err := hdkeychain.NewMaster(seedBytes, chainParams)
+	if err != nil {
+		return response, err
+	}
+
+	// Derive the extended key for account 0.
+	// This gives the path: m/0H
+	accountKey, err := masterKey.Child(hdkeychain.HardenedKeyStart + 0)
```

Comment:
There was a horrific bug in `ExtendedKey.Child` that resulted in wrong derivations, from master private keys, with a non-trivial probability. Can you please update `btcutil` and use `ExtendedKey.Derive` instead?

See related commit: https://github.com/btcsuite/btcutil/pull/182/commits/dde9e31e50c913f5fa56ff1e405dd442dd12e452

Reply:
Oh well seen!